src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -446,6 +447,56 @@ public void CleanupAfterEachTest()
             }
         }
 
+        /// <summary>
+        /// Checks if the connection string provided in the config is correctly updated for MSSQL.
+        /// If the connection string already contains the `Application Name` property, it should update the existing value along with the DataApiBuilder Application Name.
+        /// If not, it should appends the property `Application Name` to the connection string.
+        /// </summary>
+        /// <param name="databaseType">database type.</param>
+        /// <param name="providedConnectionString">connection string provided in the config.</param>
+        /// <param name="expectedUpdatedConnectionString">Updated connection string with Application Name.</param>
+        /// <param name="isHostedScenario">If Dab is hosted or OSS.</param>
+        [DataTestMethod]
+        [DataRow(DatabaseType.MSSQL, "Something;", "Something;Application Name=dab_oss_1.0.0;", false, DisplayName = "[MSSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.MySQL, "Something;", "Something;", false, DisplayName = "[MYSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.PostgreSQL, "Something;", "Something;", false, DisplayName = "[PGSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.CosmosDB_PostgreSQL, "Something;", "Something;", false, DisplayName = "[COSMOSDB_PGSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.CosmosDB_NoSQL, "Something;", "Something;", false, DisplayName = "[COSMOSDB_NOSQL]:Connection String doesn't contain Application name for DAB oss.")]
+        [DataRow(DatabaseType.MSSQL, "Something;Application Name=CustAppName;", "Something;Application Name=CustAppName,dab_oss_1.0.0;", false, DisplayName = "[MSSQL]:Connection String contains customer Application name for DAB oss.")]
+        [DataRow(DatabaseType.MSSQL, "Something1;Application Name=CustAppName;Something2;", "Something1;Application Name=CustAppName,dab_oss_1.0.0;Something2;", false, DisplayName = "[MSSQL2]:Connection String contains customer Application name for DAB oss.")]
+        [DataRow(DatabaseType.MySQL, "Something;Application Name=CustAppName;", "Something;Application Name=CustAppName;", false, DisplayName = "[MYSQL]:Connection String contains customer Application name for DAB oss.")]
```

Comment:
This doesnt seem to be a separate scenario for the other Dbs, the case where connstring is not updated at all should cover this situation as well

Reply:
this is test that No change will be made to the connection string provided by the Customer irrespective of `Application Name` property in the conn_string